jscomp/ml/printtyp.ml

```diff
@@ -1521,6 +1521,22 @@ let report_subtyping_error ppf env tr1 txt1 tr2 =
     let tr1 = List.map prepare_expansion tr1
     and tr2 = List.map prepare_expansion tr2 in
     fprintf ppf "@[<v>%a" (trace true (tr2 = []) txt1) tr1;
+    (match tr1 with 
+    | [(t1, _); (_, t2)] ->
+      let a_runtime_representation = Runtime_representation.to_runtime_representation t2 env in
+      let b_runtime_representation = Runtime_representation.to_runtime_representation t1 env in
+      a_runtime_representation |> List.iter(
+        fun a_value -> 
+          b_runtime_representation |> List.iter(
+            fun b_value -> 
+              if Runtime_representation.runtime_values_match a_value b_value then (
+                ()
+              ) 
+              else Runtime_representation.explain_why_not_matching a_value b_value 
+                |> List.iter(fun s -> fprintf ppf "@ %s" s)
+          ))
+          | _ -> ()
+    );
```

Comment:
@cristianoc the code in here is very difficult to understand, so this isn't correct, but I think it might be along the lines at least of what we'd want if we want to enhance the subtyping messages with information about the runtime representations.

A missing piece is additional information and hints about how something is configured (@ as, @ tag, etc). Haven't figured out yet how to tackle that.